File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        Stack<Integer> stack = new Stack<>();
        int answer = 0;
        for(int i = 0; i < ingredient.length; i++) {
            stack.push(ingredient[i]);
            int n = stack.size();
            if(n >= 4) {
                if(stack.get(n-4) == 1 && stack.get(n-3) == 2 && stack.get(n-2) == 3 && stack.get(n-1) == 1) {
                    answer++;
                    for(int j = 0; j <4; j++) {
                        stack.pop();
                    }
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    var answer = 0;
    const stack = [];
    
    for (let i = 0; i < ingredient.length; i++) {
        stack.push(ingredient[i]);
        const len = stack.length;
        if(len >= 4 && stack[len -1] === 1) {
            if(stack[len - 2] === 3 && stack[len - 3] === 2 && stack[len - 4] === 1) {
                for(let j = 0; j < 4; j++) stack.pop();
                answer++;
            }
        }
    }
    
    return answer;
}